tests/unit/heap.c

```diff
@@ -23,9 +23,66 @@
 
 #include <igraph.h>
 
+#include "test_utilities.inc"
+
 int main() {
 
-    /* This is not used by any functions any more,
-       no need to test it right now */
+    igraph_heap_t h;
+    igraph_integer_t i;
+    igraph_real_t *arr =(igraph_real_t*) calloc(5, sizeof(igraph_real_t));
+
+    /* igraph_heap_init, igraph_heap_destroy  */
+    printf("Test igraph_heap_init, igraph_heap_destroy\n");
+    igraph_heap_init(&h, 0);
+    igraph_heap_destroy(&h);
+    igraph_heap_init(&h, 10);
+    igraph_heap_destroy(&h);
+
+    /* igraph_heap_reserve */
+    printf("Test igraph_heap_reserve\n");
+    igraph_heap_init(&h, 0);
+    igraph_heap_reserve(&h, 10);
+    igraph_heap_reserve(&h, 5);
+    igraph_heap_destroy(&h);
+
+    /* igraph_heap_init_array*/
+    printf("Test igraph_heap_init_array\n");
+    igraph_heap_init_array(&h,arr,5);
+
+    /* igraph_heap_empty igraph_heap_size */
+    printf("Test igraph_heap_size, igraph_heap_empty\n");
+    IGRAPH_ASSERT((igraph_heap_size(&h)==5) && (!igraph_heap_empty(&h)));
```

Comment:
Note that this won't test `heap_empty` at all if `(igraph_heap_size(&h)==5)` evaluates to true (short-circuiting `&&`). The `heap_size` and `heap_empty` calls should be separated, i.e. not joined with `&&`.